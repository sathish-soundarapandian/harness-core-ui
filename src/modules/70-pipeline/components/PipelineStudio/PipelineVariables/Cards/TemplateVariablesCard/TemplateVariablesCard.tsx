/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { Card, AllowedTypes, NestedAccordionPanel } from '@wings-software/uicore'
import cx from 'classnames'
import type { NGVariable } from 'services/pipeline-ng'
import { StepType } from '@pipeline/components/PipelineSteps/PipelineStepInterface'
import { StepWidget } from '@pipeline/components/AbstractSteps/StepWidget'
import { StepViewType } from '@pipeline/components/AbstractSteps/Step'
import type { AbstractStepFactory } from '@pipeline/components/AbstractSteps/AbstractStepFactory'
import type {
  CustomVariableEditableExtraProps,
  CustomVariablesData
} from '@pipeline/components/PipelineSteps/Steps/CustomVariables/CustomVariableEditable'
import type { AllNGVariables } from '@pipeline/utils/types'
import VariableAccordionSummary from '@pipeline/components/PipelineStudio/PipelineVariables/VariableAccordionSummary'
import type { PipelineVariablesData } from '@pipeline/components/PipelineStudio/PipelineVariables/types'
import css from '../../PipelineVariables.module.scss'

export interface TemplateVariablesCardProps {
  variableVariables: NGVariable[]
  variables: NGVariable[]
  stepsFactory: AbstractStepFactory
  metadataMap: PipelineVariablesData['metadataMap']
  updateVariables(variables: NGVariable[]): void
  readonly?: boolean
  allowableTypes: AllowedTypes
}

export default function TemplateVariablesCard(props: TemplateVariablesCardProps): React.ReactElement {
  const { variableVariables, variables, metadataMap, stepsFactory, updateVariables, readonly, allowableTypes } = props

  return (
    <Card className={css.variableCard} id="Pipeline-panel">
      <NestedAccordionPanel
        noAutoScroll
        isDefaultOpen
        key={`template.variables`}
        id={`template.variables`}
        addDomId
        collapseProps={{
          keepChildrenMounted: true
        }}
        summary={<VariableAccordionSummary>Template Variables</VariableAccordionSummary>}
        details={
          <StepWidget<CustomVariablesData, CustomVariableEditableExtraProps>
            factory={stepsFactory}
            initialValues={{ variables: (variables || []) as AllNGVariables[], canAddVariable: true }}
            type={StepType.CustomVariable}
            stepViewType={StepViewType.InputVariable}
            readonly={readonly}
            allowableTypes={allowableTypes}
            onUpdate={({ variables: updatedVariables }: CustomVariablesData) => {
              updateVariables(updatedVariables)
            }}
            customStepProps={{
              formName: 'addEditTemplateCustomVariableForm',
              variableNamePrefix: 'template.variables.',
              domId: 'Template.Variables-panel',
              className: cx(css.customVariables, css.customVarPadL1, css.addVariableL1),
              // heading: <b>{getString('customVariables.title')}</b>,
              path: 'template.variables',
              hideExecutionTimeField: true,
              yamlProperties: (variableVariables as AllNGVariables[])?.map(
                variable => metadataMap[variable.value || '']?.yamlProperties || {}
              )
            }}
          />
        }
      />
    </Card>
  )
}
