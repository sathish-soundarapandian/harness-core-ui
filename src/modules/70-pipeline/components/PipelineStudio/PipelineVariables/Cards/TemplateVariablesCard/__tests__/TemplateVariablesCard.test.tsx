import React from 'react'
import { render } from '@testing-library/react'
import { noop } from 'lodash-es'
import { MultiTypeInputType } from '@wings-software/uicore'
import { TestWrapper } from '@common/utils/testUtils'
import TemplateVariablesCard, {
  TemplateVariablesCardProps
} from '@pipeline/components/PipelineStudio/PipelineVariables/Cards/TemplateVariablesCard/TemplateVariablesCard'
import factory from '@pipeline/components/PipelineSteps/PipelineStepFactory'
import type { NGVariable } from 'services/pipeline-ng'

const baseProps: TemplateVariablesCardProps = {
  variableVariables: [
    {
      name: 'var1',
      type: 'String',
      description: 'B7CvLEz5RaqKJhOukNZgoA',
      value: 'v3s1hLGDS0ucwF1X3Yek3g',
      __uuid: 'K6eF-MapS1OBUq0RWklMbA'
    },
    {
      name: 'var2',
      type: 'String',
      description: 'Ip_wDXFWRYy-2VdIR15Gsw',
      value: 'lhSYVGLTTfSNZnP4__5OXg',
      __uuid: 'E9LNEUVDQr-RiFsXwfvJUA'
    }
  ] as unknown as NGVariable[],
  variables: [
    {
      name: 'var1',
      type: 'String',
      value: '<+input>'
    },
    {
      name: 'var2',
      type: 'String',
      value: '<+input>'
    }
  ] as unknown as NGVariable[],
  metadataMap: {
    gqaH2BXmTZyxvmOFLpNoJw: {
      yamlProperties: {
        fqn: 'timeout',
        localName: 'timeout',
        variableName: 'timeout',
        aliasFQN: '',
        visible: true
      },
      yamlOutputProperties: {
        fqn: '',
        localName: ''
      },
      yamlExtraProperties: {
        properties: [],
        outputproperties: []
      }
    },
    GJn52W6kTYSI7DsqZPNPGg: {
      yamlProperties: {
        fqn: 'name',
        localName: 'name',
        variableName: 'name',
        aliasFQN: '',
        visible: true
      },
      yamlOutputProperties: {
        fqn: '',
        localName: ''
      },
      yamlExtraProperties: {
        properties: [],
        outputproperties: []
      }
    },
    lhSYVGLTTfSNZnP4__5OXg: {
      yamlProperties: {
        fqn: 'var2',
        localName: 'var2',
        variableName: 'var2',
        aliasFQN: '',
        visible: true
      },
      yamlOutputProperties: null,
      yamlExtraProperties: null
    },
    T2DTgZUuRkOERpQr_o76cA: {
      yamlProperties: {
        fqn: '',
        localName: '',
        variableName: '',
        aliasFQN: '',
        visible: false
      },
      yamlOutputProperties: {
        fqn: '',
        localName: ''
      },
      yamlExtraProperties: {
        properties: [
          {
            fqn: 'type',
            localName: 'type',
            variableName: 'type',
            aliasFQN: '',
            visible: true
          },
          {
            fqn: 'identifier',
            localName: 'identifier',
            variableName: 'identifier',
            aliasFQN: '',
            visible: true
          },
          {
            fqn: 'when',
            localName: 'when',
            variableName: 'when',
            aliasFQN: '',
            visible: true
          },
          {
            fqn: '.startTs',
            localName: '.startTs',
            variableName: '',
            aliasFQN: '',
            visible: false
          },
          {
            fqn: '.endTs',
            localName: '.endTs',
            variableName: '',
            aliasFQN: '',
            visible: false
          }
        ],
        outputproperties: [
          {
            fqn: '.output.httpUrl',
            localName: '.output.httpUrl',
            variableName: '',
            aliasFQN: '',
            visible: true
          },
          {
            fqn: '.output.httpMethod',
            localName: '.output.httpMethod',
            variableName: '',
            aliasFQN: '',
            visible: true
          },
          {
            fqn: '.output.httpResponseCode',
            localName: '.output.httpResponseCode',
            variableName: '',
            aliasFQN: '',
            visible: true
          },
          {
            fqn: '.output.httpResponseBody',
            localName: '.output.httpResponseBody',
            variableName: '',
            aliasFQN: '',
            visible: true
          },
          {
            fqn: '.output.status',
            localName: '.output.status',
            variableName: '',
            aliasFQN: '',
            visible: true
          },
          {
            fqn: '.output.errorMsg',
            localName: '.output.errorMsg',
            variableName: '',
            aliasFQN: '',
            visible: true
          },
          {
            fqn: '.output.outputVariables',
            localName: '.output.outputVariables',
            variableName: '',
            aliasFQN: '',
            visible: true
          }
        ]
      }
    },
    v3s1hLGDS0ucwF1X3Yek3g: {
      yamlProperties: {
        fqn: 'var1',
        localName: 'var1',
        variableName: 'var1',
        aliasFQN: '',
        visible: true
      },
      yamlOutputProperties: null,
      yamlExtraProperties: null
    },
    qnL_0ASKRtS0_XgGw64_Qw: {
      yamlProperties: {
        fqn: 'spec.method',
        localName: 'spec.method',
        variableName: 'method',
        aliasFQN: '',
        visible: true
      },
      yamlOutputProperties: {
        fqn: '',
        localName: ''
      },
      yamlExtraProperties: {
        properties: [],
        outputproperties: []
      }
    },
    'i50TW6K_T--8P6NFuaXPIw': {
      yamlProperties: {
        fqn: 'spec.requestBody',
        localName: 'spec.requestBody',
        variableName: 'requestBody',
        aliasFQN: '',
        visible: true
      },
      yamlOutputProperties: {
        fqn: '',
        localName: ''
      },
      yamlExtraProperties: {
        properties: [],
        outputproperties: []
      }
    },
    'pPxSQS7-RT6SjwPAPogE8Q': {
      yamlProperties: {
        fqn: 'description',
        localName: 'description',
        variableName: 'description',
        aliasFQN: '',
        visible: true
      },
      yamlOutputProperties: {
        fqn: '',
        localName: ''
      },
      yamlExtraProperties: {
        properties: [],
        outputproperties: []
      }
    },
    'IvujVQy3R-21Ms3fmtV7kQ': {
      yamlProperties: {
        fqn: 'spec.assertion',
        localName: 'spec.assertion',
        variableName: 'assertion',
        aliasFQN: '',
        visible: true
      },
      yamlOutputProperties: {
        fqn: '',
        localName: ''
      },
      yamlExtraProperties: {
        properties: [],
        outputproperties: []
      }
    },
    eWLGvZ0UQPaYKx0lIx1vJg: {
      yamlProperties: {
        fqn: 'spec.url',
        localName: 'spec.url',
        variableName: 'url',
        aliasFQN: '',
        visible: true
      },
      yamlOutputProperties: {
        fqn: '',
        localName: ''
      },
      yamlExtraProperties: {
        properties: [],
        outputproperties: []
      }
    },
    BcxVAZUQR7egzk4sci8h1g: {
      yamlProperties: {
        fqn: 'spec.delegateSelectors',
        localName: 'spec.delegateSelectors',
        variableName: 'delegateSelectors',
        aliasFQN: '',
        visible: true
      },
      yamlOutputProperties: {
        fqn: '',
        localName: ''
      },
      yamlExtraProperties: {
        properties: [],
        outputproperties: []
      }
    }
  } as any,
  stepsFactory: factory,
  updateVariables: noop,
  allowableTypes: [MultiTypeInputType.FIXED, MultiTypeInputType.RUNTIME, MultiTypeInputType.EXPRESSION]
}

describe('<TemplateVariablesCard/> tests', () => {
  test('should match snapshot', () => {
    const { container } = render(
      <TestWrapper>
        <TemplateVariablesCard {...baseProps} />
      </TestWrapper>
    )
    expect(container).toMatchSnapshot()
  })
})
