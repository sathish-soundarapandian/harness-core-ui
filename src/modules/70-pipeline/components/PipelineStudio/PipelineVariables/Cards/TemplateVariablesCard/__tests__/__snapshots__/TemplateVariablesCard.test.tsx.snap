// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TemplateVariablesCard/> tests should match snapshot 1`] = `
<div>
  <div
    class="bp3-card bp3-elevation-0 Card--card variableCard"
    id="Pipeline-panel"
  >
    <div
      class="Accordion--panel"
      data-open="false"
      data-testid="template.variables-panel"
      id="template.variables-panel"
    >
      <div
        class="Accordion--summary"
        data-testid="template.variables-summary"
      >
        <div
          class=""
        >
          <div
            class="accordionSummary"
          >
            <span
              class="bp3-icon StyledProps--main iconCollapsed"
              data-icon="accordion-collapsed"
            >
              <svg
                fill="none"
                height="12"
                viewBox="0 0 12 12"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="#F3F3FA"
                  height="11.5"
                  rx="0.75"
                  stroke="#9293AB"
                  stroke-width="0.5"
                  width="11.5"
                  x="0.25"
                  y="0.25"
                />
                <path
                  d="M10 6.571H6.571V10H5.43V6.571H2V5.43h3.429V2H6.57v3.429H10V6.57z"
                  fill="#0A3364"
                  opacity="0.7"
                />
              </svg>
            </span>
            <span
              class="bp3-icon StyledProps--main iconExpanded"
              data-icon="accordion-expanded"
            >
              <svg
                fill="none"
                height="12"
                viewBox="0 0 12 11"
                width="12"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="#F3F3FA"
                  height="10.5"
                  rx="0.75"
                  stroke="#9293AB"
                  stroke-width="0.5"
                  width="11.5"
                  x="0.25"
                  y="0.25"
                />
                <path
                  d="M10 6.143H2V5h8v1.143z"
                  fill="#0A3364"
                  opacity="0.7"
                />
              </svg>
            </span>
            <p
              class="StyledProps--font StyledProps--main StyledProps--font-variation-h6"
            >
              <p
                class="StyledProps--font StyledProps--main StyledProps--font-variation-small-semi StyledProps--color StyledProps--color-black"
              >
                Template Variables
              </p>
            </p>
          </div>
        </div>
        <div
          class="Accordion--chevron"
        />
      </div>
      <div
        class="bp3-collapse Accordion--collapse"
      >
        <div
          aria-hidden="true"
          class="bp3-collapse-body"
          style="transform: translateY(-undefinedpx);"
        >
          <div
            class="Accordion--details"
            data-testid="template.variables-details"
          >
            <div
              class="customVariables customVariables customVarPadL1 addVariableL1"
              id="Template.Variables-panel"
            >
              <div
                class="headerRow"
              >
                <div />
                <button
                  class="bp3-button Button--button StyledProps--font StyledProps--main add-variable Button--with-current-color Button--iconOnly Button--withLeftIcon Button--variation Button--variation-link Button--size Button--small"
                  type="button"
                >
                  <span
                    class="bp3-icon bp3-icon-plus StyledProps--main"
                    icon="plus"
                  >
                    <svg
                      data-icon="plus"
                      height="16"
                      viewBox="0 0 16 16"
                      width="16"
                    >
                      <desc>
                        plus
                      </desc>
                      <path
                        d="M13 7H9V3c0-.55-.45-1-1-1s-1 .45-1 1v4H3c-.55 0-1 .45-1 1s.45 1 1 1h4v4c0 .55.45 1 1 1s1-.45 1-1V9h4c.55 0 1-.45 1-1s-.45-1-1-1z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                  <span
                    class="bp3-button-text"
                  >
                    <span>
                      common.addVariable
                    </span>
                  </span>
                </button>
              </div>
              <div
                class="variableListTable variable-list-row"
              >
                <p
                  class="StyledProps--font Text--lineclamp StyledProps--main variableNameCell"
                  style="--text-line-clamp: 1;"
                >
                  <span
                    class=""
                  >
                    var1
                  </span>
                </p>
                <div
                  class="valueRow"
                >
                  <div>
                    <div
                      class="bp3-form-group variableInput"
                    >
                      
                      <div
                        class="bp3-form-content"
                      >
                        <div
                          class="Layout--horizontal StyledProps--main MultiTypeInput--mini MultiTypeInput--disabled"
                          style="width: 264px;"
                        >
                          <div
                            class="TextInput--main MultiTypeInput--input"
                          >
                            <div
                              class="bp3-input-group bp3-disabled"
                            >
                              <input
                                class="bp3-input"
                                disabled=""
                                name="variables[0].value"
                                placeholder="<+input>"
                                type="text"
                                value="<+input>"
                              />
                            </div>
                          </div>
                          <span
                            class="bp3-popover-wrapper MultiTypeInput--wrapper"
                          >
                            <span
                              class="bp3-popover-target"
                            >
                              <button
                                class="MultiTypeInput--miniBtn MultiTypeInput--RUNTIME runtimeInputButton"
                              >
                                <span
                                  class="bp3-icon StyledProps--main"
                                  data-icon="runtime-input"
                                >
                                  <svg
                                    height="12"
                                    viewBox="0 0 10 7.97"
                                    width="12"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M8.14 4L8 3.72a2.53 2.53 0 01-.12-.3l-.16-.35c-.11-.25-.22-.38-.32-.38s-.28.15-.37.44C7 3.1 7 3.07 7 3a.73.73 0 01.24-.46.66.66 0 01.42-.26c.25 0 .5.35.76 1.05l.07.17.07-.12c.47-.73.86-1.1 1.18-1.1a.71.71 0 01.29.07l-.4.42h-.11c-.24 0-.52.28-.86.83l-.09.16.07.2c.29.79.53 1.18.71 1.18s.31-.12.4-.36q.09.06.09.12t-.27.39a.74.74 0 01-.45.23c-.25 0-.51-.36-.77-1.07l-.12-.25-.1.17c-.44.76-.87 1.14-1.28 1.14a.65.65 0 01-.39-.12L6.84 5a.32.32 0 00.23.1c.25 0 .53-.25.83-.74l.17-.28zm-4-.32L1.7.59h2.47S5 .52 5.35 1a3.28 3.28 0 01.42 1.14h.31L5.94 0H0v.31l3.06 3.74L0 7.62V8h6.08l.38-2.36-.31-.07a2.43 2.43 0 01-.52 1c-.42.48-2 .41-2 .41H1.39z"
                                      fill="current-color"
                                    />
                                  </svg>
                                </span>
                              </button>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="actionButtons"
                  >
                    <div
                      class="configureButton"
                    >
                      <span
                        class="bp3-popover-wrapper bp3-dark"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <button
                            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow Button--withRightIcon"
                            id="configureOptions_var1"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="bp3-icon bp3-icon-cog StyledProps--main"
                              icon="cog"
                            >
                              <svg
                                data-icon="cog"
                                height="16"
                                viewBox="0 0 16 16"
                                width="16"
                              >
                                <desc>
                                  cog
                                </desc>
                                <path
                                  d="M15.19 6.39h-1.85c-.11-.37-.27-.71-.45-1.04l1.36-1.36c.31-.31.31-.82 0-1.13l-1.13-1.13a.803.803 0 00-1.13 0l-1.36 1.36c-.33-.17-.67-.33-1.04-.44V.79c0-.44-.36-.8-.8-.8h-1.6c-.44 0-.8.36-.8.8v1.86c-.39.12-.75.28-1.1.47l-1.3-1.3c-.3-.3-.79-.3-1.09 0L1.82 2.91c-.3.3-.3.79 0 1.09l1.3 1.3c-.2.34-.36.7-.48 1.09H.79c-.44 0-.8.36-.8.8v1.6c0 .44.36.8.8.8h1.85c.11.37.27.71.45 1.04l-1.36 1.36c-.31.31-.31.82 0 1.13l1.13 1.13c.31.31.82.31 1.13 0l1.36-1.36c.33.18.67.33 1.04.44v1.86c0 .44.36.8.8.8h1.6c.44 0 .8-.36.8-.8v-1.86c.39-.12.75-.28 1.1-.47l1.3 1.3c.3.3.79.3 1.09 0l1.09-1.09c.3-.3.3-.79 0-1.09l-1.3-1.3c.19-.35.36-.71.48-1.1h1.85c.44 0 .8-.36.8-.8v-1.6a.816.816 0 00-.81-.79zm-7.2 4.6c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
                                  fill-rule="evenodd"
                                />
                              </svg>
                            </span>
                          </button>
                        </span>
                      </span>
                    </div>
                    <div>
                      <span
                        class="bp3-popover-wrapper"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <button
                            class="bp3-button Button--button StyledProps--font StyledProps--main buttonsActions Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
                            data-testid="edit-variable-0"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="bp3-icon StyledProps--main"
                              data-icon="Edit"
                            >
                              <svg
                                height="18"
                                viewBox="0 0 16 16"
                                width="18"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  clip-rule="evenodd"
                                  d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                />
                                <path
                                  clip-rule="evenodd"
                                  d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                  opacity="0.3"
                                />
                              </svg>
                            </span>
                          </button>
                        </span>
                      </span>
                    </div>
                    <div>
                      <span
                        class="bp3-popover-wrapper"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <button
                            class="bp3-button Button--button StyledProps--font StyledProps--main buttonsActions Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
                            data-testid="delete-variable-0"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="bp3-icon StyledProps--main StyledProps--color"
                              data-icon="main-trash"
                            >
                              <svg
                                height="16"
                                viewBox="0 0 16 20"
                                width="16"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  clip-rule="evenodd"
                                  d="M1.143 8.48a1.2 1.2 0 011.189-1.36h10.649a1.2 1.2 0 011.194 1.321l-1.039 10.208a1.2 1.2 0 01-1.193 1.079H3.716a1.2 1.2 0 01-1.189-1.04L1.143 8.48zm11.837-.16H2.332l1.385 10.207h8.226L12.98 8.32zM5.916 1.594a.6.6 0 01.719-.452l4.224.965a.6.6 0 01.452.719L11 4.179l3.86.882a.6.6 0 01-.267 1.17L1.535 3.248a.6.6 0 11.267-1.17l3.805.87.309-1.354zm.86 1.62l3.056.698.175-.768-3.055-.698-.175.768z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                />
                                <path
                                  clip-rule="evenodd"
                                  d="M8 10.808a.5.5 0 01.5.5v5a.5.5 0 01-1 0v-5a.5.5 0 01.5-.5zm-2.761.012a.5.5 0 01.541.454l.436 4.981a.5.5 0 11-.996.087l-.436-4.98a.5.5 0 01.455-.542zM10.762 10.82a.5.5 0 01.455.541l-.436 4.981a.5.5 0 11-.996-.087l.435-4.98a.5.5 0 01.542-.455z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                  opacity="0.3"
                                />
                              </svg>
                            </span>
                          </button>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="variableListTable variable-list-row"
              >
                <p
                  class="StyledProps--font Text--lineclamp StyledProps--main variableNameCell"
                  style="--text-line-clamp: 1;"
                >
                  <span
                    class=""
                  >
                    var2
                  </span>
                </p>
                <div
                  class="valueRow"
                >
                  <div>
                    <div
                      class="bp3-form-group variableInput"
                    >
                      
                      <div
                        class="bp3-form-content"
                      >
                        <div
                          class="Layout--horizontal StyledProps--main MultiTypeInput--mini MultiTypeInput--disabled"
                          style="width: 264px;"
                        >
                          <div
                            class="TextInput--main MultiTypeInput--input"
                          >
                            <div
                              class="bp3-input-group bp3-disabled"
                            >
                              <input
                                class="bp3-input"
                                disabled=""
                                name="variables[1].value"
                                placeholder="<+input>"
                                type="text"
                                value="<+input>"
                              />
                            </div>
                          </div>
                          <span
                            class="bp3-popover-wrapper MultiTypeInput--wrapper"
                          >
                            <span
                              class="bp3-popover-target"
                            >
                              <button
                                class="MultiTypeInput--miniBtn MultiTypeInput--RUNTIME runtimeInputButton"
                              >
                                <span
                                  class="bp3-icon StyledProps--main"
                                  data-icon="runtime-input"
                                >
                                  <svg
                                    height="12"
                                    viewBox="0 0 10 7.97"
                                    width="12"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M8.14 4L8 3.72a2.53 2.53 0 01-.12-.3l-.16-.35c-.11-.25-.22-.38-.32-.38s-.28.15-.37.44C7 3.1 7 3.07 7 3a.73.73 0 01.24-.46.66.66 0 01.42-.26c.25 0 .5.35.76 1.05l.07.17.07-.12c.47-.73.86-1.1 1.18-1.1a.71.71 0 01.29.07l-.4.42h-.11c-.24 0-.52.28-.86.83l-.09.16.07.2c.29.79.53 1.18.71 1.18s.31-.12.4-.36q.09.06.09.12t-.27.39a.74.74 0 01-.45.23c-.25 0-.51-.36-.77-1.07l-.12-.25-.1.17c-.44.76-.87 1.14-1.28 1.14a.65.65 0 01-.39-.12L6.84 5a.32.32 0 00.23.1c.25 0 .53-.25.83-.74l.17-.28zm-4-.32L1.7.59h2.47S5 .52 5.35 1a3.28 3.28 0 01.42 1.14h.31L5.94 0H0v.31l3.06 3.74L0 7.62V8h6.08l.38-2.36-.31-.07a2.43 2.43 0 01-.52 1c-.42.48-2 .41-2 .41H1.39z"
                                      fill="current-color"
                                    />
                                  </svg>
                                </span>
                              </button>
                            </span>
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="actionButtons"
                  >
                    <div
                      class="configureButton"
                    >
                      <span
                        class="bp3-popover-wrapper bp3-dark"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <button
                            class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main Button--with-current-color Button--iconOnly Button--without-shadow Button--withRightIcon"
                            id="configureOptions_var2"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="bp3-icon bp3-icon-cog StyledProps--main"
                              icon="cog"
                            >
                              <svg
                                data-icon="cog"
                                height="16"
                                viewBox="0 0 16 16"
                                width="16"
                              >
                                <desc>
                                  cog
                                </desc>
                                <path
                                  d="M15.19 6.39h-1.85c-.11-.37-.27-.71-.45-1.04l1.36-1.36c.31-.31.31-.82 0-1.13l-1.13-1.13a.803.803 0 00-1.13 0l-1.36 1.36c-.33-.17-.67-.33-1.04-.44V.79c0-.44-.36-.8-.8-.8h-1.6c-.44 0-.8.36-.8.8v1.86c-.39.12-.75.28-1.1.47l-1.3-1.3c-.3-.3-.79-.3-1.09 0L1.82 2.91c-.3.3-.3.79 0 1.09l1.3 1.3c-.2.34-.36.7-.48 1.09H.79c-.44 0-.8.36-.8.8v1.6c0 .44.36.8.8.8h1.85c.11.37.27.71.45 1.04l-1.36 1.36c-.31.31-.31.82 0 1.13l1.13 1.13c.31.31.82.31 1.13 0l1.36-1.36c.33.18.67.33 1.04.44v1.86c0 .44.36.8.8.8h1.6c.44 0 .8-.36.8-.8v-1.86c.39-.12.75-.28 1.1-.47l1.3 1.3c.3.3.79.3 1.09 0l1.09-1.09c.3-.3.3-.79 0-1.09l-1.3-1.3c.19-.35.36-.71.48-1.1h1.85c.44 0 .8-.36.8-.8v-1.6a.816.816 0 00-.81-.79zm-7.2 4.6c-1.66 0-3-1.34-3-3s1.34-3 3-3 3 1.34 3 3-1.34 3-3 3z"
                                  fill-rule="evenodd"
                                />
                              </svg>
                            </span>
                          </button>
                        </span>
                      </span>
                    </div>
                    <div>
                      <span
                        class="bp3-popover-wrapper"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <button
                            class="bp3-button Button--button StyledProps--font StyledProps--main buttonsActions Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
                            data-testid="edit-variable-1"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="bp3-icon StyledProps--main"
                              data-icon="Edit"
                            >
                              <svg
                                height="18"
                                viewBox="0 0 16 16"
                                width="18"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  clip-rule="evenodd"
                                  d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                />
                                <path
                                  clip-rule="evenodd"
                                  d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                  opacity="0.3"
                                />
                              </svg>
                            </span>
                          </button>
                        </span>
                      </span>
                    </div>
                    <div>
                      <span
                        class="bp3-popover-wrapper"
                      >
                        <span
                          class="bp3-popover-target"
                        >
                          <button
                            class="bp3-button Button--button StyledProps--font StyledProps--main buttonsActions Button--iconOnly Button--withLeftIcon Button--variation Button--variation-icon"
                            data-testid="delete-variable-1"
                            tabindex="0"
                            type="button"
                          >
                            <span
                              class="bp3-icon StyledProps--main StyledProps--color"
                              data-icon="main-trash"
                            >
                              <svg
                                height="16"
                                viewBox="0 0 16 20"
                                width="16"
                                xmlns="http://www.w3.org/2000/svg"
                              >
                                <path
                                  clip-rule="evenodd"
                                  d="M1.143 8.48a1.2 1.2 0 011.189-1.36h10.649a1.2 1.2 0 011.194 1.321l-1.039 10.208a1.2 1.2 0 01-1.193 1.079H3.716a1.2 1.2 0 01-1.189-1.04L1.143 8.48zm11.837-.16H2.332l1.385 10.207h8.226L12.98 8.32zM5.916 1.594a.6.6 0 01.719-.452l4.224.965a.6.6 0 01.452.719L11 4.179l3.86.882a.6.6 0 01-.267 1.17L1.535 3.248a.6.6 0 11.267-1.17l3.805.87.309-1.354zm.86 1.62l3.056.698.175-.768-3.055-.698-.175.768z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                />
                                <path
                                  clip-rule="evenodd"
                                  d="M8 10.808a.5.5 0 01.5.5v5a.5.5 0 01-1 0v-5a.5.5 0 01.5-.5zm-2.761.012a.5.5 0 01.541.454l.436 4.981a.5.5 0 11-.996.087l-.436-4.98a.5.5 0 01.455-.542zM10.762 10.82a.5.5 0 01.455.541l-.436 4.981a.5.5 0 11-.996-.087l.435-4.98a.5.5 0 01.542-.455z"
                                  fill="currentColor"
                                  fill-rule="evenodd"
                                  opacity="0.3"
                                />
                              </svg>
                            </span>
                          </button>
                        </span>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
